src/pint/models/parameter.py

```diff
@@ -1348,26 +1349,35 @@ def __init__(
         aliases=[],
     ):
         self.is_mask = True
+        # {key_name: (keyvalue parse function, keyvalue length)}
+        # Move this to some other places.
         self.key_identifier = {
             "mjd": (lambda x: time.Time(x, format="mjd").mjd, 2),
-            "freq": (float, 2),
+            "freq": (lambda x: u.Quantity(x, u.MHz, copy=False), 2),
             "name": (str, 1),
-            "tel": (str, 1),
+            "tel": (lambda x: get_observatory(str(x)).name, 1),
         }
-        if key_value is None:
-            key_value = []
-        elif not isinstance(key_value, list):
+        if not isinstance(key_value, (list, tuple)):
             key_value = [key_value]
 
         # Check key and key value
-        if key is not None and key.lower() in self.key_identifier.keys():
-            key_info = self.key_identifier[key.lower()]
-            if len(key_value) != key_info[1]:
-                errmsg = "key " + key + " takes " + key_info[1] + " element."
-                raise ValueError(errmsg)
-
+        key_value_parser = str
+        if key is not None:
+            if key.lower() in self.key_identifier.keys():
+                key_info = self.key_identifier[key.lower()]
+                if len(key_value) != key_info[1]:
+                    errmsg = f"key {key} takes {key_info[1]} element(s)."
+                    raise ValueError(errmsg)
+                key_value_parser = key_info[0]
+            else:
+                if not key.startswith("-"):
+                    raise ValueError(
+                        "A key/flag(excpet for 'mjd', 'freq', "
+                        "'name', 'tel') needs a leading '-'."
```

Comment:
Perhaps "A JUMP keyed to a TOA flag requires a leading '-'. Legal JUMP keywords that don't require a leading '-' are MJD, FREQ, NAME, TEL."

Reply:
I am a little bit hesitant to put the word "JUMP" here. Since this is in the maskParameter, which is not only for JUMPs. Is this for keys general enough?

Reply:
Oh, I see. Well I'm just trying to make it as clear as possible for some user who loads a par file and wants to be clearly told why it is failing

Reply:
Do all maskParameters respect MJD, FREQ, etc.?

Reply:
I am not sure if anyone used ECORR EFAC like this. Those are quite general, so I believe one can.

Reply:
In PINT, yes, all EFAC/EQUAD/JUMP/etc. use the same kinds of selection criteria.